homeassistant/components/tplink/sensor.py

```diff
@@ -152,8 +201,11 @@ def __init__(
         self._attr_unique_id = (
             f"{legacy_device_id(self.device)}_{self.entity_description.key}"
         )
+        self.timezone = self.coordinator.hass.config.time_zone
 
     @property
-    def native_value(self) -> float | None:
+    def native_value(self) -> float | datetime.datetime | None:
         """Return the sensors state."""
-        return async_emeter_from_device(self.device, self.entity_description)
+        if self.entity_description.value_fn is None:
+            return None
```

Comment:
This `None` check would be nice to avoid by forcing passing a `value_fn` in the definition, which does not work due to the aforementioned issue with "Attributes without a default cannot follow attributes with one". Ideas to make this cleaner are welcome!